pkg/metrics/workqueue.go

```diff
@@ -0,0 +1,126 @@
+package metrics
+
+import (
+	"k8s.io/client-go/util/workqueue"
+	k8smetrics "k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+// Package prometheus sets the workqueue DefaultMetricsFactory to produce
+// prometheus metrics. To use this package, you just have to import it.
```

Comment:
We need to doc that this is a port of upstream code that landed in a future version. I assume there was a lot of copy/pasting here?

Reply:
comment updated @adambkaplan